server/task/task-memory/src/test/java/org/apache/james/task/eventsourcing/TaskAggregateTest.java

```diff
@@ -85,6 +85,30 @@ void givenInProgressTaskEmitEventWhenUpdateAdditionalInformationCommand() {
             .containsExactly(AdditionalInformationUpdated.apply(ID, history.getNextEventId(), new MemoryReferenceWithCounterTask.AdditionalInformation(3, timestamp)));
     }
 
+    @Test
+    void givenInProgressTaskWithOneOlderUpdateEventEmitEventWhenUpdateAdditionalInformationCommand() {
+        History history = buildHistory(
+            eventId -> Created.apply(ID, eventId, new MemoryReferenceWithCounterTask((counter) -> Task.Result.COMPLETED), HOSTNAME),
```

Comment:
is the lambda ever called?

Reply:
the one to create the history ? Yes
the one from the task ? No

Reply:
the one from the task.
We should try to minimize useless code in tests.

Reply:
It used the same task definition as in the previous test. IMHO it's the minimum task definition.
We could throw a NotImplemented exception but it would be uglier and maybe even longer.